src/agent.rs

```diff
@@ -262,11 +282,21 @@ impl AgentThread {
             // alignment 1 is always correct
             .alloc(unsafe { &Layout::from_size_align_unchecked(self.max_sr_data_len, 1) })?;
         loop {
-            let sz = self
+            for i in header_buf.as_mut_slice() {
+                *i = 0_u8;
+            }
+            let (sz, imm) = self
                 .inner
                 .qp
                 .receive_sge(&[&mut header_buf, &mut data_buf])
                 .await?;
+            if header_buf.as_slice() == CLEAN_STATE && imm.is_some() {
```

Comment:
exchange these two condition, for non-imm request we don't need to compare slice.